enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/RegisteredPipeBuilder.scala

```diff
@@ -94,6 +95,22 @@ class RegisteredPipeBuilder(fallback: PipeBuilder,
         val toSlot = pipeline(to)
         ExpandIntoRegisterPipe(source, fromSlot.offset, relSlot.offset, toSlot.offset, dir, LazyTypes(types), pipeline)(id)
 
+      case VarExpand(_, IdName(fromName), dir, projectedDir, types, IdName(toName), IdName(relName), VarPatternLength(min, max), expansionMode, predicates) =>
+        // TODO: This is not right!
+        if(predicates.nonEmpty)
+          throw new CantCompileQueryException("does not handle varexpand with predicates")
```

Comment:
Very different. I have been working on solving this, but it can be done as a separate step. I'll push what I have so you can see what I'm thinking about.